### PR TITLE
s3_logging - migrate to boto3

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Migrate the s3_logging module to boto3
- Also add support for check_mode
- Re-enable tests (with retries on missing ACLs)

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

s3_logging

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: plugins/modules/s3_logging.py [hunk 1]
```diff
@@ -59,27 +59,68 @@
...
+    if module.check_mode:
         return True
 
+    updated_acl = dict(current_acl)
+    updated_grants = list(current_grants)
+    updated_grants.append(required_grant)
+    updated_acl['Grants'] = updated_grants
+    del updated_acl['ResponseMetadata']
+    try:
+        connection.put_bucket_acl(Bucket=target_bucket, AccessControlPolicy=updated_acl)
```
Comment (alinabuzachis) [inline plugins/modules/s3_logging.py:118]: aws_retry?

File: plugins/modules/s3_logging.py [hunk 2]
```diff
@@ -89,29 +130,36 @@ def enable_bucket_logging(connection, module):
...
+            bucket_logging = camel_dict_to_snake_dict(bucket_logging)
+            module.exit_json(changed=changed, **bucket_logging)
 
-    except S3ResponseError as e:
-        module.fail_json(msg=to_native(e))
+        if module.check_mode:
+            module.exit_json(changed=True)
 
-    module.exit_json(changed=changed)
+        result = connection.put_bucket_logging(
```
Comment (alinabuzachis) [inline plugins/modules/s3_logging.py:149]: aws_retry?

File: plugins/modules/s3_logging.py (hunk 1 above)
Comment (tremble) [inline plugins/modules/s3_logging.py:118]:
```suggestion
        connection.put_bucket_acl(aws_retry=True, Bucket=target_bucket, AccessControlPolicy=updated_acl)
```

File: plugins/modules/s3_logging.py [hunk 3]
```diff
@@ -89,29 +130,36 @@ def enable_bucket_logging(connection, module):
...
+            Bucket=bucket_name,
```
Comment (tremble) [inline plugins/modules/s3_logging.py:150]:
```suggestion
            aws_retry=True,
            Bucket=bucket_name,
```

File: plugins/modules/s3_logging.py (hunk 1 above)
Comment (tremble) [reply plugins/modules/s3_logging.py]: Nice catch, thanks

**Review by alinabuzachis — APPROVED** [review]

**Comment by tremble** [comment]
Has an approval.  tests pass.  Merging.

**Pull request merged**